pkgs/swiftgen/swift2objc/lib/src/parser/parsers/declaration_parser.dart

```diff
@@ -0,0 +1,35 @@
+import '../../ast/_core/interfaces/declaration.dart';
+import '../_core/parsed_symbol.dart';
+import '../_core/utils.dart';
+import 'declaration_parsers/class_parser.dart';
+import 'declaration_parsers/method_parser.dart';
+
+class DeclarationParser {
+  final ParsedSymbolsMap parsedSymbolsMap;
+  final String parsedSymbolId;
```

Comment:
You're never actually using to `parsedSymbolId` directly. The only place it's used is the `parsedSymbol` getter. So if you take the `ParsedSymbol` directly in the constructor then you could remove that getter.

The getter is a bit weird anyway. You're doing a map lookup every time you need to refer to the `parsedSymbol`.

Reply:
I refactored things into functions as we discussed and fixed this issue.